src/context.rs

```diff
@@ -449,41 +437,18 @@ impl Context {
 
     /// Emits a single event.
     pub fn emit_event(&self, event: EventType) {
-        if self
-            .debug_logging
-            .try_read()
-            .ok()
-            .map(|inner| inner.is_some())
-            == Some(true)
         {
-            self.send_log_event(event.clone()).ok();
-        };
+            let lock = tokio::task::block_in_place(|| self.debug_logging.blocking_read());
```

Comment:
I never benchmarked how fast `block_in_place` is, but since it moves the worker to another thread, it might be too slow to call it on every event, even if debug logging is disabled.

Since we never hold a lock over an await point, it would be OK to do blocking_read without a block_in_place, or even completely use a non-async mutex. [The documentation](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html) even advises using a non-async mutex in this case since it's faster. (IIUC if we use a non-async mutex then the compiler will even prevent us from accidentally holding a lock over an await point, so that would prob. be better than using blocking_read with a synchronous mutex).

Reply:
I just noticed it's an RwLock not a Mutex, but this still applies.